stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/plugin/state/StateRegistry.kt

```diff
@@ -68,4 +71,16 @@ public class StateRegistry internal constructor(
         channels.clear()
         threads.clear()
     }
+
+    internal companion object {
+        private var instance: StateRegistry? = null
+
+        internal fun getOrCreate(chatDomainImpl: ChatDomainImpl): StateRegistry {
+            return instance ?: StateRegistry(chatDomainImpl).also { stateRegistry ->
+                instance = stateRegistry
+            }
+        }
+
+        internal fun get(): StateRegistry? = instance
```

Comment:
Do we need this?

Reply:
Same thing with the above explanation.

Reply:
👍 consider moving `nonNull` check here as in comment above